subprojects/plugins/src/main/java/org/gradle/api/internal/java/usagecontext/ConfigurationVariantMapping.java

```diff
@@ -58,6 +60,9 @@ public void addAction(Action<? super ConfigurationVariantDetails> action) {
     }
 
     public void collectUsageContexts(final ImmutableCollection.Builder<UsageContext> outgoing) {
+        if (!outgoingConfiguration.isTransitive()) {
+            SingleMessageLogger.nagUserWith("Publication ignores 'transitive = false' at configuration level.", "", "Consider using 'transitive = false' at the dependency level if you need this to be published.", "", DeprecatedFeatureUsage.Type.USER_CODE_INDIRECT);
```

Comment:
:x: This is only half related to this PR, but there is already a `validate()` method in the class. So this should probably go in there.
BUT, that method is never called? I think it was broken in refactoring and we miss test coverage for `...doesn't declare any capability`.

Reply:
From what I can see in the history, this code was never used. Will delete it as I can't see how it can work. We have no way of differentiating between the _main_ variant, which can have no capabilities (aka the default capability) and a feature variant which should have a non default capability.

Reply:
Makes sense. Thanks for checking.